src/session_impl.cpp

```diff
@@ -5661,7 +5655,7 @@ namespace aux {
 		m_dht_settings = settings;
 	}
 
-	void session_impl::set_dht_state(dht::dht_state const& state)
+	void session_impl::set_dht_state(dht::dht_state state)
 	{
 		m_dht_state = state;
```

Comment:
any value that has a name is an lvalue and has to be explicitly moved. another way of thinking about it is that you could, if you wanted to, still refer to the ``state`` value after the assignment, so the compiler cannot decide to move it away from under you. Since only you can make the promise that you won't use the value again later, only you can ask for the move.